app/index.js

```diff
@@ -579,13 +579,21 @@ ipcMain.on("download-list-request", (e, action, downloadId) => {
 ipcMain.on("set-permissions", (_, permissionObject) => {
     permissions = permissionObject
 })
-ipcMain.on("set-spelllang", (_, lang) => {
+ipcMain.on("set-spelllang", (_, langOptions) => {
+    langOptions = new Set(
+        langOptions.split(",").map(lang => {
+            if (lang === "system") {
+                return app.getLocale()
+            }
+
+            return lang
+        })
+    )
+
     sessionList.forEach(ses => {
-        if (lang === "system") {
-            session.fromPartition(ses).setSpellCheckerLanguages([])
-        } else {
-            session.fromPartition(ses).setSpellCheckerLanguages([lang])
-        }
+        session.fromPartition(ses).setSpellCheckerLanguages(
+            Array.from(langOptions)
```

Comment:
Why do we need Array.from here? Can't we just build an array to begin with? I don't see the function of building a new set, if we are only going to use it as an array.

Reply:
I started with a Set because I wanted to get rid of duplicates like `set spelllocale=en_US, system`. If the system already has the default language as "en_US" it would be duplicated. I don't know how Electron would process it but I rather not risk it. The same applies to passing the Set to Electron function

Reply:
Makes sense, but I would move that to happen when setting the option, so that duplicates aren't stored at all, instead of just not being used here.